SECURITY.md

```diff
@@ -109,6 +109,11 @@ does not trust is considered a vulnerability:
   the correct use of Node.js APIs.
 * The unavailability of the runtime, including the unbounded degradation of its
   performance.
+* Memory leaks qualify as vulnerabilities when all of the following criteria are met:
+  * The API is being correctly used
+  * The API doesn't have a warning against its usage in a production environment
+  * The API is on stable (2.0) status
+  * The memory leak is significant, causing a DoS fast or in a user-uncontrolled space (for instance, on HTTP parsing)
```

Comment:
I feel that we still need more restrictions to avoid flagging all the existing memory leaks as vulnerabilities just because they weren't reported as vulnerabilities (for example, https://github.com/nodejs/node/issues/54614 probably satisfies all the criteria and it can be used in server handlers). As long as a buggy API is in the wrong place, it can cause a DoS fast, and it doesn't have to be a memory leak, so it seems to be a very slippery slope.

Reply:
I feel like https://github.com/nodejs/node/issues/54614 is still fit in the new criteria, which is not expected to be so.

Reply:
Also I think it makes sense to include [asymmetric resource consumption](https://cwe.mitre.org/data/definitions/405.html) into the criteria: if it costs not significantly less resources to cause resource exhaustion on the server, then it isn't really a practical attack